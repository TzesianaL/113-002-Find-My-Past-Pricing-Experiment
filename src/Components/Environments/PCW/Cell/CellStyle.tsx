import styled from 'styled-components'

export const CellStyle = styled.div<{
  shaded: boolean
  listType: boolean
}>`
  display: flex;
  flex-direction: column;
  text-align: ${(props) => (props.listType ? 'left' : 'center')};
  align-items: ${(props) => (props.listType ? 'flex-start' : 'center')};
  justify-content: center;
  color: ${(props) => props.theme.pcwCell?.textColor || '#000000'};
  background: ${(props) =>
    props.shaded
      ? '#DCDCDC'
      : props.theme.pcwCell?.backgroundColor || '#ffffff'};
  @media (max-width: 600px) {
    min-width: 180px;
  }
  min-width: 100px;
  height: 90px;
  border: 1px solid;
  border-color: lightgray;
  word-wrap: break-word;
  font-weight: bolder;
  font-size: small;
`
