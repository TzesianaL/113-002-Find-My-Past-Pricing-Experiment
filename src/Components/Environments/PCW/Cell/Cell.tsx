import React, { FC } from 'react'
import { CSSProperties } from 'styled-components'
import { CellStyle } from './CellStyle'

export interface CellProps {
  shaded?: boolean
  style?: CSSProperties
  listType?: boolean
}

const Cell: FC<CellProps> = ({ shaded, style, children, listType }) => {
  return (
    <CellStyle
      style={style}
      shaded={shaded || false}
      listType={listType || false}
    >
      {children}
    </CellStyle>
  )
}

export default Cell
