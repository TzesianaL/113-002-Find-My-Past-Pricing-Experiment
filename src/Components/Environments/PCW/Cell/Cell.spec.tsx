import React from 'react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import * as renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import Cell from './Cell'

describe('<Cell/>', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Cell />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should grey out background when shaded prop is true', () => {
    const { container } = render(<Cell shaded>Test</Cell>)
    expect(container.querySelector('div')).toHaveStyleRule(
      'background',
      '#DCDCDC'
    )
  })
  it('should not grey out background when shaded prop is not provided', () => {
    const { container } = render(<Cell>Test</Cell>)
    expect(container.querySelector('div')).toHaveStyleRule(
      'background',
      '#ffffff'
    )
  })
})
