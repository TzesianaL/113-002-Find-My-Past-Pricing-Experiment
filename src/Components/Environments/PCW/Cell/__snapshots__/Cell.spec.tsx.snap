// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell/> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #000000;
  background: #ffffff;
  min-width: 100px;
  height: 90px;
  border: 1px solid;
  border-color: lightgray;
  word-wrap: break-word;
  font-weight: bolder;
  font-size: small;
}

@media (max-width:600px) {
  .c0 {
    min-width: 180px;
  }
}

<div
  className="c0"
/>
`;
